libcxx/CMakeLists.txt

```diff
@@ -424,6 +428,8 @@ set(LIBCXX_INSTALL_INCLUDE_DIR "${CMAKE_INSTALL_INCLUDEDIR}/c++/v1" CACHE STRING
     "Path where target-agnostic libc++ headers should be installed.")
 set(LIBCXX_INSTALL_RUNTIME_DIR "${CMAKE_INSTALL_BINDIR}" CACHE STRING
     "Path where built libc++ runtime libraries should be installed.")
+set(LIBCXX_INSTALL_MODULE_DIR "${CMAKE_INSTALL_INCLUDEDIR}/../modules/c++/v1" CACHE STRING
```

Comment:
Following @ruoso 's comment in https://github.com/llvm/llvm-project/pull/75741#issuecomment-1861199593, I would suggest making this something much less pretty than `modules/c++/v1` so as to not give the impression that folks can rely on this as a stable name.

Reply:
Suggestions from our live review just now:

```
$prefix/usr/include/c++/v1/__private_modules/
$prefix/usr/modules/c++/v1/
$prefix/usr/share/libc++/v1/ (or whatever was agreed upon on the Discourse RFC)
$prefix/usr/share/c++/v1/
```

Reply:
@jwakely What do you think about this? Any opinion? This path technically doesn't matter because it should always be extracted from the JSON file, but it would be nice if both GCC and Clang had semi-consistent paths.

The JSON file itself is always going to be next to the dylib in  e.g.`$prefix/usr/lib/libc++.module.json` (and in theory always accessed via something like `clang -print-file-name=libc++.module.json`).

Reply:
usr is a $prefix, so probably intended as
```
$prefix/include/c++/v1/__private_modules/
$prefix/include/c++/v1/
$prefix/usr/share/libc++/v1/
etc
```

I see very few things in my usr/include that aren't textual inclusion for C or C++. There's some evidence that fortran may put mod files there, but it looks like rust puts similar information in share/cargo, which isn't a choice for C++.